src/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -297,26 +316,48 @@ class ThetaStarPathPlanner : public PathPlanner
     void resetAndInitializeMemberVariables(const Rectangle &navigable_area,
                                            const std::vector<ObstaclePtr> &obstacles);
 
-    // if close to destination then return no path
+    /**
+     * Returns a collision-free hash for the given Coordinate
+     *
+     * @param coord Coordinate to hash
+     *
+     * @return hash of the Coordinate
+     */
+    unsigned long hashCoordinate(const Coordinate &coord) const;
+
+    /**
+     * Returns a collision-free hash for the given a pair of Coordinate
+     *
+     * @param coord1 Coordinate 1 of pair to hash
+     * @param coord2 Coordinate 2 of pair to hash
+     *
+     * @return hash of the pair of Coordinates
+     */
+    unsigned long hashCoordinatePair(
+        const ThetaStarPathPlanner::Coordinate &coord1,
+        const ThetaStarPathPlanner::Coordinate &coord2) const;
```

Comment:
Why did we elect to have a separate hash function like this and then a map from a `long` instead of just making a `CoordinatePair` struct and then making that hashable? 

Similarly for the `Coordinate` datatype.

Reply:
I thought this implementation would be best for performance. We also have to guarantee collision-free

Reply:
collision free just makes hash-based datastructures faster because no linear search is required if there's a collision. I don't see how having a separate function has any implication on the collision-free guarantee.

Reply:
The hashing must be perfectly collision-free because a collision would mean that the map lookup gets the wrong value. For example, if `h = hash(i) == hash(j)`, then `unblocked_map[hash(i)] ==unblocked_map[hash(j)]`, even if `i` is unblocked and `j` is blocked. I wasn't sure conventional hashing would be perfectly collision free

Reply:
I'm a bit confused, you've effectively implemented a hash function already, why not just create a hash instead of creating a `long` and then hashing that? It seems like it would be functionally equivalent?

Reply:
Sorry, this is bad naming. This is not a hash. This is a way to make the map keys unsigned long instead of with Coordinate or pair of Coordinates, which speeds up accesses. Changed these functions to `computeMapKey`

Reply:
Isn't this effectively the same thing as a hash? (_trying to avoid holding up this pr for just this comment, so please feel free to ping me on slack_)

Reply:
I think the points made about the hashing stuff aren't that what is currently being done is a bad hash algorithm, more that it's slightly confusing because the same algorithm hasn't been implemented just as a hash operator for the datatype.

Reply:
It's not a hash function: https://en.wikipedia.org/wiki/Hash_function A hash function maps data of arbitrary size to a fixed size value. In this case, we're mapping a domain of known size (Coordinates or Coordinate Pairs) to a range of essentially unlimited size (unsigned long)

Reply:
I'm pretty sure "fixed size value" you're thinking of here can represent exactly the same range of values as an `unsigned long`?